ui/apps/platform/src/Containers/Collections/CollectionForm.tsx

```diff
@@ -182,19 +182,8 @@ function CollectionForm({
     });
 
     useEffect(() => {
-        if (saveError) {
-            return;
-        }
-        // We need to manually validate the values onChange here since Formik update the values
-        // before validation and the declarative state has `isValid === true` and `isValidation == false`
-        // at the same time that the actual value object is invalid.
-        validationSchema
-            .validate(values)
-            .then(() => onFormChange(values))
-            .catch(() => {
-                /* Validation failed, do not propagate change */
-            });
-    }, [values, saveError, onFormChange]);
+        onFormChange(values);
```

Comment:
We need to send dry run requests even when the form is invalid to get correct error handling (e.g. the user goes to create a collection, skips entering a name, and configures a collection rule), so we can discard this logic and just propagate all form changes. Since we debounce at the request level now, this won't cause a flood of requests to hit the server.